### PR TITLE
Improvements to Profile 2 Super Immersive Mode

### PR DESCRIPTION
**Pull request by DrBeef**
This PR finishes off the Profile 2 of Super Immersive mode that I wasn't entirely happy with. The downside is that it requires the knowledge/discovery of an additional bit of game-specific config, the upside is that it **greatly** improves the experience, bringing it a lot closer to a proper VR level of quality.

With the extra item of configuration set correctly the view will be stereo-correct at all angles, whereas previously (and also without the extra config) it was only truly stereo correct when looking forwards.

`vr_immersive_eye_indicator` - String - needs to be a pair of numbers, effectively a register offset (much like the 6DoF register offset) and the index into the register (since each "register" is a Vec4f). So for example, Mario Kart 7 would use "87,2".

Fundamentally, the register/index specified refers to a float that represents which "eye" is being drawn for.. having looked at a few games now, there appears to be a pretty consistent behaviour that there is always a register that describes this, and so can be used as an indicator of which eye the particular draw call is being made for. With this knowledge, we can perform our own stereo separation ensuring it is correctly aligned with the direction the user is looking. In order to achieve this, if this is configured the stereo slider value is actually passed to the emulator as set to 0.1% to effectively apply (almost) no stereo separation (it still has to apply something, even tiny, so we can get the -ve or +ve value to tell us which eye is being drawn), once we know which eye we are drawing, we move the camera left or right depending on the eye we calculate based on the register contents.

Whilst it sounds over complicated, the majority of this PR is actually just a bit of code tidy up and configuration (and I re-positioned the virtual screen slightly better).. the actual core part of this extra functionality is the block of code at line 958 in src/video_core/rasterizer_accelerated.cpp

Finally, if this extra bit of config isn't specified, then Super Immersive Profile 2 just works exactly as it currently does.

I'll try and write a guide to discovering these register offsets as it isn't exactly a trivial process, and for the eye indicator value, it is easiest using a debugger and the desktop version of citra!

### CODE REVIEW

**Comment by DrBeef** [comment]
I just committed the extra change that adds the heuristic search for the registers that means no additional config for Profile 2 is now necessary (unless it doesn't work for a particular game). 

Few points:

- It'll run for a period of time during the early stage of running a game, it isn't particularly resource heavy and will stop after a short while once it has settled on a register
- I have added a register value of -1, which is the default now, for this new "auto-detect" mode
- I've tried to document the search algorithm with comments about what each bit is doing, it's fairly straightforward, awards points to registers that meet certain criteria, over time the best register is likely the correct one.
- It both the register for the view matrix transformation and the register for the left/right eye indicator, so no configuration needed at all

Hopefully this is acceptable to merge! it certainly makes using the super immersive mode a lot simpler.

File: src/video_core/rasterizer_accelerated.cpp [hunk 1]
```diff
@@ -22,6 +24,8 @@ static Common::Vec3f LightColor(const Pica::LightingRegs::LightColor& color) {
     return Common::Vec3u{color.r, color.g, color.b} / 255.0f;
 }
 
+constexpr float VR_IPD  = 0.065f;
```
Comment (lvonasek) [inline src/video_core/rasterizer_accelerated.cpp:27]: You could get realtime IPD by calculating the distance between left and right eye positions.

**Comment by DrBeef** [comment]
Final additional commit to improve the location of the virtual screen and fix a rendering issue in OOT and MM on profile 1

**Comment by DrBeef** [comment]
Just want to add, this PR is definitely ready to merge now.

**Pull request closed without merging**